marimo/_convert/ipynb.py

```diff
@@ -608,22 +619,29 @@ def on_ref(
     return new_sources
 
 
-def transform_cell_metadata(
-    sources: list[str], metadata: list[dict[str, Any]]
-) -> list[str]:
+def bind_cell_metadata(
+    sources: list[str], metadata: list[dict[str, Any]], hide_flags: list[bool]
+) -> list[CodeCell]:
     """
-    Handle cell metadata, such as tags or cell IDs.
+    One-time transformation that binds sources and (ipynb) metadata into CodeCell objects.
+
+    This marks the boundary between source-only transformations and cell-level transformations.
+
+    - If "hide-cell" is present in the tags, the cell is marked hidden (and removed)
+    - Remaining tags (if any) are inserted as a comment at the top of the source.
```

Comment:
We have a few other options like disable etc..
Are there other analogs?

Reply:
I can have a closer look. I think for now we could start to enumerate what we anticipate we could handle, then we snapshot the "cell_metadata.ipynb" notebook with a case of the specific one we expect.